src/tags/object/RichText/table.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * TableText component is a component which renders JSON data.
+ *
+ * The item data is expected to be JSON, which is then rendered in a speech
+ * bubble format.
+ *
+ * This injects a renderTableValue function into the RichTextPieceView component.
+ * Which allows us to convert the JSON data into HTML to allow us to directly
+ * reuse the annotation features already built into the RichTextPieceView component.
+ * Note that the focus on this is to minimize the editing / code copying from
+ * RichTextPieceView, and hence we added two new props to the RichTextPieceView:
+ *
+ * - valueToComponent: a function that converts the JSON value to a React component
+ * - alwaysInline: a boolean that indicates whether the component should always be inline
+ *   since otherwise the component is rendered in separate IFrame which makes
+ *   styling / LaTex rendering much more difficult.
+ */
+
+import React from 'react';
+import { inject, observer } from 'mobx-react';
+import { types } from 'mobx-state-tree';
+import { cn } from '../../../utils/bem';
+
+import './RichText.styl';
+import { RichTextModel } from './model';
+import { HtxRichText } from './view';
+
+const renderTableValue = (val) => {
+  let conversations = [];
+
+  try {
+    conversations = JSON.parse(val);
+  } catch (e) {
+    // Display red text box with error message
+    const errClass = cn('richtext', { elem: 'error-box' });
+    const errMsg = `Couldn't parse JSON: ${e.message}`;
+
+    console.error(errMsg);
+    return <div className={errClass}>{errMsg}</div>;
+  }
+
+  const itemClass = cn('richtext', { elem: 'table-item' });
+
+  const rowElems = conversations.map((conversation, index) => {
+    const question = conversation[0];
```

Comment:
**Question**:   Trying to compare this to the Thread Interactions imported into the BetterMath project - seems like this works?   

```
["Help me solve this", "Of course! Let's start by looking at the problem."]
```

question = "Help me solve this"
answer = "Of course! Let's start by looking at the problem."

Do you anticipate we'll need to add any validation steps to ensure we can render correctly if the RichTextTable is leveraged?

Reply:
We will expect a specific JSON format (e.g. a list of 2-tuple of question and answers) - though the field will be generated from the import pipeline (e.g. in BetterMath case, this is generated from the SQL import pipeline), so it is kinda already validated :-)